### PR TITLE
Decomp scrcmd_15 -> scrcmd_party

### PR DESCRIPTION
**Pull request by Admiral-Fish**

### CODE REVIEW

File: src/scrcmd_party.c [hunk 1]
```diff
@@ -0,0 +1,811 @@
...
+    FieldSystem *fsys = ctx->fsys;
+    u16 slot = ScriptGetVar(ctx);
+    u16 *species = ScriptGetVarPointer(ctx);
+
+    POKEMON *pokemon = GetPartyMonByIndex(SavArray_PlayerParty_get(fsys->savedata), slot);
+    if (GetMonData(pokemon, MON_DATA_IS_EGG, 0) == 0) {
+        *species = GetMonData(pokemon, MON_DATA_SPECIES, 0);
+    }
+    else {
+        *species = 0;
```
Comment (PikalaxALT) [inline src/scrcmd_party.c:49]:
```suggestion
        *species = SPECIES_NONE;
```

File: src/scrcmd_party.c [hunk 2]
```diff
@@ -0,0 +1,811 @@
...
+
+    u16 *slot = ScriptGetVarPointer(ctx);
+    u16 *mine = ScriptGetVarPointer(ctx);
+
+    POKEMON *pokemon = GetPartyMonByIndex(SavArray_PlayerParty_get(fsys->savedata), *slot);
+    u16 pokemonID = GetMonData(pokemon, MON_DATA_OTID, 0);
+    u16 playerID = PlayerProfile_GetTrainerID(profile);
+
+    if (pokemonID == playerID) {
+        *mine = 0;
```
Comment (PikalaxALT) [inline src/scrcmd_party.c:68]:
```suggestion
        *mine = FALSE;
```

File: src/scrcmd_party.c [hunk 3]
```diff
@@ -0,0 +1,811 @@
...
+    }
+    else {
+        *mine = 1;
```
Comment (PikalaxALT) [inline src/scrcmd_party.c:71]:
```suggestion
        *mine = TRUE;
```

File: src/scrcmd_party.c [hunk 4]
```diff
@@ -0,0 +1,811 @@
...
+}
+
+BOOL ScrCmd_MonHasMove(SCRIPTCONTEXT *ctx) {
+    FieldSystem *fsys = ctx->fsys;
+    u16 *hasMove = ScriptGetVarPointer(ctx);
+    u16 move = ScriptGetVar(ctx);
+    u16 slot = ScriptGetVar(ctx);
+
+    POKEMON *pokemon = GetPartyMonByIndex(SavArray_PlayerParty_get(fsys->savedata), slot);
+    *hasMove = 0;
```
Comment (PikalaxALT) [inline src/scrcmd_party.c:115]:
```suggestion
    *hasMove = FALSE;
```

File: src/scrcmd_party.c [hunk 5]
```diff
@@ -0,0 +1,811 @@
...
+    if (GetMonData(pokemon, MON_DATA_IS_EGG, 0) != 0) {
+        return FALSE;   
+    }
+
+    if (GetMonData(pokemon, MON_DATA_MOVE1, 0) == move || GetMonData(pokemon, MON_DATA_MOVE2, 0) == move ||
+        GetMonData(pokemon, MON_DATA_MOVE3, 0) == move || GetMonData(pokemon, MON_DATA_MOVE4, 0) == move) {
+        *hasMove = 1;
```
Comment (PikalaxALT) [inline src/scrcmd_party.c:122]:
```suggestion
        *hasMove = TRUE;
```

File: src/scrcmd_party.c [hunk 6]
```diff
@@ -0,0 +1,811 @@
...
+BOOL ScrCmd_GetPartySlotWithMove(SCRIPTCONTEXT *ctx) {
+    FieldSystem *fsys = ctx->fsys;
+    u16 *slot = ScriptGetVarPointer(ctx);
+    u16 move = ScriptGetVar(ctx);
+    u8 i;
+
+    u8 partyCount = GetPartyCount(SavArray_PlayerParty_get(fsys->savedata));
+    for (i = 0, *slot = 6; i < partyCount; i++) {
+        POKEMON *pokemon = GetPartyMonByIndex(SavArray_PlayerParty_get(fsys->savedata), i);
+        if (GetMonData(pokemon, MON_DATA_IS_EGG, 0) != 0) {
```
Comment (PikalaxALT) [inline src/scrcmd_party.c:137]:
```suggestion
        if (GetMonData(pokemon, MON_DATA_IS_EGG, 0)) {
```

File: src/scrcmd_party.c [hunk 7]
```diff
@@ -0,0 +1,811 @@
...
+}
+
+BOOL ScrCmd_MonGetLevel(SCRIPTCONTEXT *ctx) {
+    FieldSystem *fsys = ctx->fsys;
+    u16 *level = ScriptGetVarPointer(ctx);
+    u16 slot = ScriptGetVar(ctx);
+
+    POKEMON *pokemon = GetPartyMonByIndex(SavArray_PlayerParty_get(fsys->savedata), slot);
+    *level = 0;
+    if (GetMonData(pokemon, MON_DATA_IS_EGG, 0) == 0) {
```
Comment (PikalaxALT) [inline src/scrcmd_party.c:186]:
```suggestion
    if (!GetMonData(pokemon, MON_DATA_IS_EGG, 0)) {
```

File: src/scrcmd_party.c [hunk 8]
```diff
@@ -0,0 +1,811 @@
...
+}
+
+BOOL ScrCmd_GetMonTypes(SCRIPTCONTEXT *ctx) {
+    FieldSystem *fsys = ctx->fsys;
+    u16 *type1 = ScriptGetVarPointer(ctx);
+    u16 *type2 = ScriptGetVarPointer(ctx);
+    u16 slot = ScriptGetVar(ctx);
+
+    POKEMON *pokemon = GetPartyMonByIndex(SavArray_PlayerParty_get(fsys->savedata), slot);
+    *type1 = GetMonData(pokemon, MON_DATA_TYPE_1, 0);
```
Comment (PikalaxALT) [inline src/scrcmd_party.c:322]:
```suggestion
    *type1 = GetMonData(pokemon, MON_DATA_TYPE_1, NULL);
``` and everywhere else

File: src/scrcmd_party.c [hunk 9]
```diff
@@ -0,0 +1,811 @@
...
+    u16 slot = ScriptGetVar(ctx);
+
+    POKEMON *pokemon = GetPartyMonByIndex(SavArray_PlayerParty_get(fsys->savedata), slot);
+    if (GetMonData(pokemon, MON_DATA_IS_EGG, 0) != 0) {
+        *moveCount = 0;
+        return FALSE;
+    }
+
+    u8 count = 0;
+    if (GetMonData(pokemon, MON_DATA_MOVE1, 0) != 0) {
```
Comment (PikalaxALT) [inline src/scrcmd_party.c:446]:
```suggestion
    if (GetMonData(pokemon, MON_DATA_MOVE1, NULL) != MOVE_NONE) {
```

**Review by PikalaxALT — CHANGES_REQUESTED** [review]
In general:
- Arguments which expect a pointer but are passed 0, should be passed NULL
- Except where required to match, BOOL return values should be written accordingly
- Values which are taken to be boolean, even when they're not BOOL, should be set using TRUE or FALSE as appropriate.

**Comment by Admiral-Fish** [comment]
`GetMonData(pokemon, MON_DATA_FATEFUL_ENCOUNTER, NULL) == TRUE`
This was required to match

`GetMonData(pokemon, MON_DATA_HP, NULL) != 0`
I left this as is as HP is not a BOOL.

**Review by PikalaxALT — APPROVED** [review]

**Pull request merged**